added deprecated warning for ctype solar-x/solar-y conversion (#6901)
sunpy/sunpy: changelog/6901.bugfix.rst

Author: how about depreciated?

> added a deprecated warning message when the ``ctype solar-x/solar-y`` conversion happens in `sunpy/map/mapbase.py`.
> It will also help in discovering other map source dependents of this conversion, which can be fixed.

Reviewer: ```suggestion
We have deprecated the conversion of WCS ``ctype `` values of "solar-x", "solar_x" within `~sunpy.map.Map`.
```
